openfisca_core/simulations.py

```diff
@@ -114,35 +129,299 @@ def data_storage_dir(self):
                 ).format(self._data_storage_dir).encode('utf-8'))
         return self._data_storage_dir
 
-    @property
-    def holder_by_name(self):
-        warnings.warn(
-            u"The simulation.holder_by_name attribute has been deprecated. "
-            u"Please use entity.get_holder instead. "
-            u"Using simulation.holder_by_name may negatively impact performances",
-            Warning
-            )
+    # ----- Methods to access parameters ----- #
 
-        result = {}
-        for entity in self.entities.itervalues():
-            result.update(entity._holders)
-        return result
+    def parameters_at(self, instant, use_baseline = False):
+        if isinstance(instant, periods.Period):
+            instant = instant.start
+        assert isinstance(instant, periods.Instant), "Expected an Instant (e.g. Instant((2017, 1, 1)) ). Got: {}.".format(instant)
+        if use_baseline:
+            return self._get_baseline_parameters_at_instant(instant)
+        return self._get_parameters_at_instant(instant)
+
+    def _get_parameters_at_instant(self, instant):
+        parameters_at_instant = self._parameters_at_instant_cache.get(instant)
+        if parameters_at_instant is None:
+            parameters_at_instant = self.tax_benefit_system.get_parameters_at_instant(instant)
+            self._parameters_at_instant_cache[instant] = parameters_at_instant
+        return parameters_at_instant
+
+    def _get_baseline_parameters_at_instant(self, instant):
+        baseline_parameters_at_instant = self._baseline_parameters_at_instant_cache.get(instant)
+        if baseline_parameters_at_instant is None:
+            baseline_parameters_at_instant = self.tax_benefit_system._get_baseline_parameters_at_instant(
+                instant = instant,
+                traced_simulation = self if self.trace else None,
+                )
+            self.baseline_parameters_at_instant_cache[instant] = baseline_parameters_at_instant
+        return baseline_parameters_at_instant
+
+    # ----- Calculation methods ----- #
 
     def calculate(self, variable_name, period, **parameters):
```

Comment:
> You may be puzzled by the "cycle detection" part. This mechanism is certainly too complex, and could be challenged. However, this is not the goal of this PR, so be careful not to get dragged into it 🐉.

Reply:
Should we open an issue about it ?

Reply:
I can open the issue and dump my thoughts on the topic Friday.